pkg/notes/template.go

```diff
@@ -0,0 +1,250 @@
+	Stable              []string
+	Beta                []string
+	Alpha               []string
+	StagingRepositories []string
+	CliImprovements     []string
+	Misc                []string
+}
+
+const notesTemplate = `## What’s New (Major Themes)
+
+<!-- Add themes from Comms Blog here -->
+
+## Known Issues
+
+<!-- Add issues from known issues bucket (known-issues-bucket.Md) here -->
+
+## Urgent Upgrade Notes
+
+### (No, really, you MUST read this before you upgrade)
+{{range .UpgradeNotes}}{{if .Header}}
+#### {{.Header}}
+
+{{range .Content}}{{.}}
+{{end}}{{end}}{{end}}
+## Deprecations and Removals
+
+<!--
+Add it in the format
+- Component
+  - deprecation
+  - removal
+-->
+
+## Metrics Changes
+
+{{range .Metrics.Changes}}{{.}}
+{{end}}
+### Added metrics
+
+{{range .Metrics.Added}}{{.}}
+{{end}}
+### Removed metrics
+
+{{range .Metrics.Removed}}{{.}}
+{{end}}
+### Deprecated/changed metrics
+
+{{range .Metrics.Deprecated}}{{.}}
+{{end}}
+## Notable Features
+
+### Stable
+
+{{range .Features.Stable}}{{.}}
+{{end}}
+### Beta
+
+{{range .Features.Beta}}{{.}}
+{{end}}
+### Alpha
+
+{{range .Features.Alpha}}{{.}}
+{{end}}
+### Staging Repositories
+
+{{range .Features.StagingRepositories}}{{.}}
+{{end}}
+### CLI Improvements
+
+{{range .Features.CliImprovements}}{{.}}
+{{end}}
+### Misc
+
+{{range .Features.Misc}}{{.}}
+{{end}}
+## API Changes
+
+{{range .APIChanges}}{{.}}
+{{end}}
+## Other notable changes
+{{range .Other}}{{if .Header}}
+### {{ .Header }}
+
+{{range .Content}}{{.}}
+{{end}}{{end}}{{end}}
+## Dependencies
+
+<!-- Add by hand -->
+`
+
+func CreateFromTemplate(logger log.Logger, notes ReleaseNotes) (*Template, error) {
+	res := &Template{}
+	for _, note := range notes {
+		for _, sig := range note.SIGs {
+
+			definedSIG, ok := availableSIGs[sig]
+			if !ok {
+				level.Warn(logger).Log("msg", "unable to find ID for SIG "+sig)
+				continue
+			}
+			i := definedSIG.id
+
+			if note.ActionRequired {
+				res.UpgradeNotes[i].setHeaderIfNeeded(definedSIG.name)
+				res.UpgradeNotes[i].Content = markdownAppend(res.UpgradeNotes[i].Content, note)
+
+			} else if wordIn(note, "metrics") {
+				res.Metrics.classifyMetricsAppend(note)
+
+			} else if wordIn(note, "api") {
+				res.APIChanges = markdownAppend(res.APIChanges, note)
+
+			} else if note.Feature {
+				res.Features.classifyFeaturesAppend(note)
+
+			} else {
+				res.Other[i].setHeaderIfNeeded(definedSIG.name)
+				res.Other[i].Content = markdownAppend(res.Other[i].Content, note)
+			}
+		}
+	}
+
+	return res, nil
+}
+
+func (m Metrics) classifyMetricsAppend(note *ReleaseNote) {
+	if wordIn(note, "added") {
+		m.Added = markdownAppend(m.Added, note)
+
+	} else if wordIn(note, "removed") {
+		m.Removed = markdownAppend(m.Removed, note)
+
+	} else if wordIn(note, "deprecated") {
+		m.Deprecated = markdownAppend(m.Deprecated, note)
+
+	} else {
+		m.Changes = markdownAppend(m.Changes, note)
+	}
+}
+
+func (f Features) classifyFeaturesAppend(note *ReleaseNote) {
```

Comment:
If the note says "moving feature X from alpha to beta" it would keep the note in Alpha rather than put it in Beta. This might be a harder problem...